inference-engine/tests/functional/plugin/shared/include/behavior/layout.hpp

```diff
@@ -0,0 +1,132 @@
+//
+#pragma once
+
+#include <typeindex>
+#include <string>
+#include <vector>
+#include <memory>
+#include <tuple>
+#include <gtest/gtest.h>
+#include <ngraph/node.hpp>
+#include <ngraph/function.hpp>
+#include <ie_plugin_config.hpp>
+#include <ngraph/function.hpp>
+#include <ngraph_functions/subgraph_builders.hpp>
+#include "gtest/gtest.h"
+#include "common_test_utils/common_utils.hpp"
+#include "common_test_utils/test_common.hpp"
+
+#include "functional_test_utils/skip_tests_config.hpp"
+#include "functional_test_utils/plugin_cache.hpp"
+#include "functional_test_utils/blob_utils.hpp"
+#include "functional_test_utils/precision_utils.hpp"
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+#include "ngraph_functions/pass/convert_prc.hpp"
+
+typedef std::tuple<
+            InferenceEngine::Precision,             // Network precision
+            std::string,                            // Target device name
+            std::map<std::string, std::string>,     // Target config
+            InferenceEngine::Layout,                // Layout
+            std::vector<size_t>>                    // InputShapes
+LayoutParams;
+
+class LayoutTest : public CommonTestUtils::TestsCommon,
+                       public ::testing::WithParamInterface<LayoutParams> {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<LayoutParams> obj) {
+        InferenceEngine::Precision netPrecision;
+        std::string targetDevice;
+        InferenceEngine::Layout layout;
+        std::vector<size_t> inputShapes;
+        std::map<std::string, std::string> configuration;
+        std::tie(netPrecision, targetDevice, configuration, layout, inputShapes) = obj.param;
+        std::ostringstream result;
+        result << "netPRC=" << netPrecision.name() << "_";
+        result << "targetDevice=" << targetDevice << "_";
+        result << "layout=" << layout << "_";
+        if (!inputShapes.empty()) {
+            for (auto &Item : inputShapes) {
+                result << "inputShapes=" << Item << "_";
+            }
+        }
+        return result.str();
+    }
+
+    void SetUp() override {
+        std::tie(netPrecision, targetDevice, configuration, layout, inputShapes) = this->GetParam();
+        function = ngraph::builder::subgraph::make2InputSubtractIn(inputShapes, netPrecision);
+    }
+
+    void TearDown() override {
+        if ((targetDevice == CommonTestUtils::DEVICE_GPU) || (!configuration.empty())) {
+            PluginCache::get().reset();
+        }
+        function.reset();
+    }
+
+    std::shared_ptr<InferenceEngine::Core> ie = PluginCache::get().ie();
+    InferenceEngine::Layout layout;
+    std::vector<size_t> inputShapes;
+    std::shared_ptr<ngraph::Function> function;
+    InferenceEngine::Precision netPrecision;
+    std::string targetDevice;
+    std::map<std::string, std::string> configuration;
+};
+
+inline bool checkLayout(InferenceEngine::Layout layout, std::vector<size_t> inputShapes) {
```

Comment:
Please, separate this file into 2 parts - `hpp` and `cpp`. And move `checkLayout` to the cpp